GVRf/Framework/src/org/gearvrf/GVRSceneObject.java

```diff
@@ -239,15 +239,18 @@ public String getName() {
     }
 
     /**
-     * Set the (optional) name of the object.
+     * Set the (optional) name of the object. Scene object names are not needed:
+     * they are only for the application's convenience.
      * 
-     * Scene object names are not needed: they are only for the application's
-     * convenience.
-     * 
-     * @param name
-     *            Name of the object.
+     * @param name Name of the object.
+     * @throws IllegalArgumentException if input parameter is null or empty.
      */
     public void setName(String name) {
+        if (name == null || name.trim().isEmpty()) {
```

Comment:
Hi All.

I think we may need to make an agreement on these cases. Let's discuss about it tomorrow and if it is really necessary, I will write a new String check API which is similar to String.trim() but without generate a new sub String which I think is redundant.